kobo/apps/help/models.py

```diff
@@ -43,6 +43,9 @@ class InAppMessage(AbstractMarkdownxModel):
     valid_from = models.DateTimeField(default=EPOCH_BEGINNING)
     valid_until = models.DateTimeField(default=EPOCH_BEGINNING)
     last_editor = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    # We do not want to use a generic foreign key or tightly couple this model
```

Comment:
If I understand our opinion on generic foreign key, what we dislike is that it points to a django content type which itself is in the database. That adds complexity with no apparent benefit to us. We know the model's name. It never changes. We prefer to reference the model by name instead of a on the dynamic generated ID from content types.  Or to say this in a more practical way - gfk's break under certain conditions, such as transferring data from one Django DB to another. There is no guarantee that the content type ID's would line up consistently.

We remain at the risk of it losing the relationship if we did intentionally change the model name/label without a data migration. But in this case, it's fine. Go ahead and loose the connection for some old in app message. Better than a database hitting constraints and erroring out hard.

I'm 100% for this concept. But we do deviate from Django's design. Let's add more detail here (or somewhere?) explaining why we created our own solution to something that comes out of the box with Django. We may end up using it again later.

@jnm I'd be curious on your opinion of this approach.